### DISCUSSION TITLE
Change STORY_INDEX_PATH for assets hosted on a different url

### DISCUSSION THREAD

**Discussion by AckermannJan** [storybookjs/storybook#29054] [Help]
### Summary

Hey all, I am updating our current storybook 6 to 8. In our server architecture, we are hosting our static files on a different URL than where we actually visit the page. So, for example:
`https://static.cdn.COMPANY.de` but visiting storybook would be `https://www.COMPANY.de/storybook`.

Using `viteFinal`, I was already able to change most of the path to the correct URL. The only thing missing is how Storybook fetches the `index.json` formerly `stories.json`. Storybook always tries to fetch the `index.json` from a relative path, so from `https://www.COMPANY.de/storybook/index.json`, but this does not exist in our case; instead, it should load it from `https://static.cdn.COMPANY.de/index.json`.

Already in version 6, we had to do some hacky CI/CD stuff, replacing strings in the build result and searching for STORY_INDEX_PATH in the code. This is no longer possible with Storybook 8 since this variable does not exist after the build.

Is anyone aware of a way to change the STORY_INDEX_PATH to also be configurable or any other workaround?

### Additional information

This is our main.ts

```ts
import type { StorybookConfig } from "@storybook/vue3-vite";
import {loadEnv, mergeConfig, UserConfig} from "vite";
import path from "path";

const config: StorybookConfig = {
  stories: [
    '../lib/components/**/*.stories.ts',
    '../stories/*.mdx'
  ],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@storybook/addon-interactions"
  ],
  framework: {
    name: "@storybook/vue3-vite",
    options: {},
  },
  async viteFinal(config, { configType }) {
    process.env = { ...process.env, ...loadEnv(configType, process.cwd(), '') };

    let publicPath = process.env.VUE_APP_PUBLIC;
    if (process.env.STORYBOOK_SSC_PUBLIC_PATH) {
      publicPath = `${process.env.STORYBOOK_SSC_PUBLIC_PATH}`;
    }
    return mergeConfig(config, {
      base: publicPath,
      css: {
        preprocessorOptions: {
          scss: {
            additionalData: '@import "./lib/styles/utils";',
          },
        }
      },
      resolve: {
        alias: {
          '~ress/ress': path.resolve(__dirname, '../node_modules/ress/ress.css'),
          '~': path.resolve(__dirname, '../'),
          '@': path.resolve(__dirname, '../src/'),
        },
      },
    } as UserConfig);
  },
};
export default config;

### Create a reproduction

_No response_

**Comment by Marklb** [comment]
I don't know how you are hosting, so this may not be possible in your environment, but I am just suggesting a potential alternative. I don't know how to solve this the way you are doing it, at the moment.

Could the program that resolves connections to the url `https://www.COMPANY.de/storybook`, proxy the connections to `https://static.cdn.COMPANY.de`. So, nothing in the Storybook app would need to know that they are from the static cdn. I have done things like that with Nginx and IIS, but I don't do it enough to list possible side-effects or reason not to do it that way.
